docker/docker.go

```diff
@@ -148,6 +148,40 @@ func (d *ImageManager) BuildImage(dockerfileDirPath, name string, stdoutWriter i
 	return nil
 }
 
+// BuildImageFromStream builds a docker image using a tar file stream containing a Dockerfile
+// A io.Writer can be given for log messages; if it implments io.Closer, it will be closed when done.
+func (d *ImageManager) BuildImageFromStream(stream io.Reader, name string, stdoutWriter io.Writer) error {
+	bio := dockerclient.BuildImageOptions{
+		Name:         name,
+		NoCache:      true,
+		InputStream:  stream,
+		OutputStream: stdoutWriter,
+	}
+
+	for _, envVar := range []string{"http_proxy", "https_proxy", "no_proxy"} {
+		for _, name := range []string{strings.ToLower(envVar), strings.ToUpper(envVar)} {
```

Comment:
Yeah, I just copied this code from the other function.
I wonder if we should make that call this…